src/components/button/button.tsx

```diff
@@ -160,8 +161,8 @@ export const EuiButton: FunctionComponent<Props> = ({
         href={href}
         target={target}
         rel={secureRel}
-        ref={buttonRef}
-        {...rest}>
+        ref={buttonRef as Ref<HTMLAnchorElement>}
```

Comment:
TS isn't intelligent enough to follow the impact of the union's discrimination through the rest of the function. We do the same hack at https://github.com/elastic/eui/blob/651c8b399f7a57128b5bf8050305bc2ae9c2d69d/src/components/progress/progress.tsx#L88

@thompsongl found an issue in the TS repo related to this, I was unable to find the link in my chat history with him.